composables/useCollectionAttributes.ts

```diff
@@ -0,0 +1,64 @@
+import type { NFT, Attribute } from '@/types'
+
+export const useCollectionAttributes = (collectionId: ComputedRef<string | undefined>) => {
+  const { data: nftAttributesData, refetch } = useGraphql({
+    queryName: 'nftAttributesListByCollection',
+    variables: {
+      id: collectionId.value,
+    },
+    disabled: computed(() => !collectionId.value),
+  })
+
+  watch(collectionId, () => {
+    refetch({
+      id: collectionId.value,
+    })
+  })
+
+  const nftsList = computed<NFT[]>(() => {
+    return (nftAttributesData.value as unknown as { nfts: NFT[] })?.nfts || []
```

Comment:
add type to `useQuery <{ nfts: NFT[] }>`

Reply:
updated